odkx-src/tables-intents.rst

```diff
@@ -0,0 +1,243 @@
+
+        // Iterate through the query results, rendering list items
+        for (var i = 0; i < barcodeResultSet.getCount(); i++) {
+
+            // Creates the item space and stores the row ID in it
+            var item = $('<li>');
+            item.attr('id', barcodeResultSet.getRowId(i));
+            item.attr('rowId', barcodeResultSet.getRowId(i));
+            item.attr('class', 'item_space');
+
+            // Display the barcodeScanner name
+            var name = barcodeResultSet.getData(i, 'itemName');
+            if (name === null || name === undefined) {
+                name = 'unknown name';
+            }
+            item.text(name);
+
+            // Creates arrow icon
+            var chevron = $('<img>');
+            chevron.attr('src', odkCommon.getFileAsUrl('config/assets/img/little_arrow.png'));
+            chevron.attr('class', 'chevron');
+            item.append(chevron);
+
+            // Add the item to the list
+            $('#list').append(item);
+
+            // Don't append the last one to avoid the fencepost problem
+            var borderDiv = $('<div>');
+            borderDiv.addClass('divider');
+            $('#list').append(borderDiv);
+        }
+        if (i < barcodeResultSet.getCount()) {
+            setTimeout(resumeFn, 0, i);
+        }
+    };
+
+    var cbFailure = function(error) {
+        console.log('barcode getViewData CB error : ' + error);
+    };
+
+Deploy changes to the device. The Survey form will not be a requirement for this test since we earlier hooked up the button in the list view. Select the census form and click on the button to launch a barcode. This launches the barcode reader enabling you to capture the information on the barcode. Immediately this is done, control switches back to the activity responsible for launching the intent, presenting you with the results from the scan. The results are now available for further processing as desired.
+
+Example :program:`Chrome` inspect console output is shown below:
+
+.. code-block:: console
+
+    > {
+        "dispatchStruct": "{\"action\":0,\"htmlPath\":\"barcodeScnner_list\",\"actionTypeKey\":\"launchBarcode\"}",
+        "action": "com.google.zxing.client.android.SCAN",
+        "jsonValue": {
+            "status": -1,
+            "result": {
+                "SCAN_RESULT": "6801315196662",
+                "SCAN_RESULT_FORMAT": "EAN_13"
+            }
+        }
+    }
+
+Sample Screenshots are shown below:
+
+.. image:: /img/tables-sample-app/tables-barcode-button.png
+  :alt: ODK-X Tables barcode scan before
+
+.. image:: /img/tables-sample-app/tables-barcode-results.png
+  :alt: ODK-X Tables barcode scan results
+
+.. note::
+    If clicking on the button does not launch any activity to scan a barcode, it is possible that no barcode reader application exists one the device or the barcode with the zing package. Install a `zxing scanner` for example, :program:`QR & Barcode Reader` from TeaCapps.
```

Comment:
Is zing package right?

Reply:
`com.google.zxing.client.android.SCAN` is the intent so zxing package. Thanks